gap/main.gi

```diff
@@ -110,10 +116,18 @@ function (mat, options)
     Info(InfoGauss, 2, "Preparation");
 
     ##Preparation: Init and chopping the matrix mat
-
-    copyMat := MutableCopyMat(mat);
-    ConvertToMatrixRepNC(copyMat);
-    tmp := GAUSS_ChopMatrix(galoisField, copyMat, a, b);
+    if isChopped then
+        copyMat := mat;
```

Comment:
For the sake of uniform behaviour I would also copy mat for now. Or we do #241 now and also make `DoEch..` destructive by default.